ocaml/typing/persistent_env.ml

```diff
@@ -78,20 +78,38 @@ type can_load_cmis =
   | Can_load_cmis
   | Cannot_load_cmis of Lazy_backtrack.log
 
-type pers_struct = {
-  ps_is_param: bool;
-  ps_crcs: Import_info.t array;
-  ps_filename: string;
-  ps_visibility: Load_path.visibility;
+(* Data relating directly to a .cmi *)
+type import = {
+  imp_is_param : bool;
+  imp_impl : Impl.t;
+  imp_sign : Subst.Lazy.signature;
+  imp_filename : string;
+  imp_visibility: Load_path.visibility;
+  imp_crcs : Import_info.Intf.t array;
+  imp_flags : Cmi_format.pers_flags list;
 }
 
 (* If a .cmi file is missing (or invalid), we
    store it as Missing in the cache. *)
-type 'a pers_struct_info =
+type import_info =
   | Missing
-  | Found of pers_struct * 'a
+  | Found of import
+
+type binding =
+  | Static of Compilation_unit.t (* Bound to a static constant *)
+
+(* Data relating to an actual referenceable module, with a signature and a
+   representation in memory. This is currently a placeholder, since such
+   modules have no more associated data than imports, but this will soon
+   change with parameterised modules, in which each import may correspond
+   to any number of different instances, each of which is a different
+   in-memory value. *)
+type pers_struct = unit
+
+type 'a pers_struct_info = pers_struct * 'a
```

Comment:
Maybe just make this a record with one field, and then add more fields once there are parameterized modules.

Reply:
Done.